### PR TITLE
Only publish files that have changed since the last publish

### PR DESCRIPTION
**Pull request by jmhooper**
This commit replaces `publish.sh` with a slightly more sophisticated ruby script in `publish.rb`. This script downloads the MD5 hashes of the objects in the buckets and compares them to the MD5 hashes of the local files before uploading. It uses this information to determine if a file has changed since the files were last published, and only re-uploads files that have a different checksum.

### CODE REVIEW

File: publish.rb [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+  end
+end
+
+puts "Compressing local files"
+
+def should_compress_file?(filename)
+  filename.match(GZIP_REGEX)
+end
+
+GZIP_REGEX = /\.html$|\.css$|\.js$|\.json$|\.svg$/
```
Comment (jmhooper) [inline publish.rb]: This needs to move up above line 62 before it's used

**Comment by jmhooper** [comment]
The publish script now create redirects. This is no longer WIP and is ready for review.

**Comment by wslack** [comment]
Either @brandocalrissian or @jseppi can review

**Comment by wslack** [comment]
@jseppi do you know how to deploy this for staging only so we can test it?

**Comment by jseppi** [comment]
Not yet, but will look into it tomorrow!

**Pull request merged**